File: Games/Alex/src/de/hsh/alexander/actor/player/PacMan1.java
package de.hsh.alexander.actor.player;

import common.actor.Direction;
import de.hsh.alexander.actor.ResourcePaths;

import java.util.HashMap;

public final class PacMan1 extends PacMan {

    private static HashMap<String, Direction> createKeyMap() {
        HashMap<String, Direction> pacMan1KeyMap = new HashMap<>();
        pacMan1KeyMap.put( "Up", Direction.Up );
        pacMan1KeyMap.put( "Down", Direction.Down );
        pacMan1KeyMap.put( "Left", Direction.Left );
        pacMan1KeyMap.put( "Right", Direction.Right );
        return pacMan1KeyMap;
    }

    public PacMan1( double x, double y ) {
        super(x, y, createKeyMap(), ResourcePaths.Actor.Player.PacMan.pacman1Pictures );
    }

}

File: Games/Alex/src/de/hsh/alexander/actor/player/PacMan1.kt
package de.hsh.alexander.actor.player

import common.actor.Direction
import de.hsh.alexander.actor.ResourcePaths
import javafx.scene.input.KeyCode
import java.util.*

class PacMan1(x: Double, y: Double)
    : PacMan(x, y, createKeyMap(), ResourcePaths.Actor.Player.PacMan.pacman1Pictures) {
    companion object {
        private fun createKeyMap(): HashMap<KeyCode, Direction> {
            val pacMan1KeyMap = HashMap<KeyCode, Direction>()
            pacMan1KeyMap[KeyCode.UP] = Direction.Up
            pacMan1KeyMap[KeyCode.DOWN] = Direction.Down
            pacMan1KeyMap[KeyCode.LEFT] = Direction.Left
            pacMan1KeyMap[KeyCode.RIGHT] = Direction.Right
            return pacMan1KeyMap
        }
    }
}
